### PR TITLE
fix(color-contrast): check bg on fg contrast for thin text-shadows

### PR DESCRIPTION
**Pull request by WilcoFiers**
The initial text-shadow implementation overlooked that the shadow color can have the luminance between that of the foreground and background. Now, if you have a dark text, light background and a thin shadow that sits in between those two colors, axe will grab the contrast of the foreground and background color.

Related to that, the shadowOutlineEmMax threshold is bumped to .2em. I've showed a few people the .15em example and most seemed okay with comparing background and foreground for a shadow of that width.

This change also addressed some "false positive" edge cases, making the rule in general slightly more permissive when it comes to text-shadow.

Closes issue: #3157

### CODE REVIEW

File: lib/checks/color/color-contrast-evaluate.js [hunk 1]
```diff
@@ -71,14 +71,16 @@ export default function colorContrastEvaluate(node, options, virtualNode) {
...
-    // Thin shadows can pass either by contrasting with the text color
-    // or when contrasting with the background.
     shadowColor = [...shadowColors, bgColor].reduce(flattenShadowColors);
-    const bgContrast = getContrast(bgColor, shadowColor);
-    const fgContrast = getContrast(shadowColor, fgColor);
-    contrast = Math.max(bgContrast, fgContrast);
-    contrastContributor =
-      bgContrast > fgContrast ? 'shadowOnBgColor' : 'fgOnShadowColor';
+    // Compare shadow, bgColor, textColor. Check passes if any is sufficient
+    const fgBgContrast = getContrast(bgColor, fgColor);
```
Comment (WilcoFiers) [inline lib/checks/color/color-contrast-evaluate.js:76]: This is the critical change here. Considering fgColor / bgColor contrast even if there's a thin shadow.

File: lib/checks/color/color-contrast.json [hunk 2]
```diff
@@ -17,7 +17,7 @@
       }
     },
     "pseudoSizeThreshold": 0.25,
-    "shadowOutlineEmMax": 0.1
+    "shadowOutlineEmMax": 0.2
```
Comment (WilcoFiers) [inline lib/checks/color/color-contrast.json:20]: After asking a few people I discovered the current threshold was slightly too strict. Increasing this number means the slightly larger shadows will be allowed to serve as text outlines, giving more opportunities for it to pass the contrast ratio.

File: test/integration/rules/color-contrast/text-shadows.html [hunk 3]
```diff
@@ -303,220 +320,433 @@
 <section class="s1" id="b1">
   <div class="r0"><i id="sample1">Sample</i></div>
   <div class="r1">
-    <i>b1-r1-i1</i> <i>b1-r1-i2</i> <i>b1-r1-i3</i> <i>b1-r1-i4</i>
+    <i id="b1-r1-i1">b1-r1-i1</i>
```
Comment (WilcoFiers) [inline test/integration/rules/color-contrast/text-shadows.html:323]: I added IDs to make these easier to work with.

**Review by WilcoFiers — COMMENTED** [review]
To see the results locally go to http://localhost:9876/test/integration/rules/color-contrast/text-shadows.html

The script on that page adds icons to what passes and what fails.

**Review by straker — APPROVED** [review]

**Pull request merged**